src/java.security.jgss/share/classes/sun/security/jgss/wrapper/NativeGSSContext.java

```diff
@@ -370,12 +372,36 @@ public void dispose() throws GSSException {
         srcName = null;
         targetName = null;
         delegatedCred = null;
-        if (pContext != 0) {
-            pContext = cStub.deleteContext(pContext);
+
+        if (pContext != 0 && cleanable != null) {
+            cleanable.clean();
             pContext = 0;
```

Comment:
Here and below: should pContext be set to 0 before calling cleanable.clean() to make sure cleanable.clean() is not called twice? Also should there be some synchronized block?

Reply:
> Also should there be some synchronized block?

I thought about the synchronization issue.  It looks like this class is not multiple-threads safe, and thus the upper layer will take care of the synchronization.  So I did not add synchronization in this class update.

> should pContext be set to 0 before calling cleanable.clean() to make sure cleanable.clean() is not called twice? 

Cleanable.clean() will work one time only, no matter how many time it is called.  But it looks like better to make a change here.